ckanext/harvest/plugin.py

```diff
@@ -36,14 +37,37 @@ class Harvest(p.SingletonPlugin, DefaultDatasetForm):
 
     ## IPackageController
 
+    def _analyze_to_run_externally(data_dict):
```

Comment:
Judging from the name of this method, I wouldn't expect it to mutate the data_dict passed in. Seems like it should just return a boolean.

Reply:
I [updated the function name to be clear](https://github.com/avdata99/ckanext-harvest/commit/7c37d28048660ebe977bed574444deffc860f0d8).

Reply:
Maybe [this](https://github.com/avdata99/ckanext-harvest/commit/633d269a2c8e37060914a46c863ac8f99b319333) is cleaner

We do not overwrite the dict, we just get the right status:
```
def _get_harvest_source_state(data_dict):
        cfg_str = data_dict.get('config', '{}')
        try:
            cfg = json.loads(cfg_str)
        except Exception as e:
            log.error('Failed to read harvest source configuration: {} {}'.format(cfg_str, str(e)))
            return data_dict

        ret = 'inactive' if cfg.get('harvester_app', 'CKAN') != 'CKAN' else 'active'
        
        return ret
```